services/httpd/handler.go

```diff
@@ -185,26 +200,67 @@ func NewHandler(c Config) *Handler {
 		},
 		Route{ // Ping
 			"ping",
-			"GET", "/ping", false, true, h.servePing,
+			"GET", "/ping", false, true, authWrapper(h.servePing),
 		},
 		Route{ // Ping
 			"ping-head",
-			"HEAD", "/ping", false, true, h.servePing,
+			"HEAD", "/ping", false, true, authWrapper(h.servePing),
 		},
 		Route{ // Ping w/ status
 			"status",
-			"GET", "/status", false, true, h.serveStatus,
+			"GET", "/status", false, true, authWrapper(h.serveStatus),
 		},
 		Route{ // Ping w/ status
 			"status-head",
-			"HEAD", "/status", false, true, h.serveStatus,
+			"HEAD", "/status", false, true, authWrapper(h.serveStatus),
 		},
 		Route{
 			"prometheus-metrics",
-			"GET", "/metrics", false, true, promhttp.Handler().ServeHTTP,
+			"GET", "/metrics", false, true, authWrapper(promhttp.Handler().ServeHTTP),
 		},
 	}...)
 
+	// When PprofAuthEnabled is enabled, create debug/pprof endpoints with the
```

Comment:
What if `!h.Config.AuthEnabled && h.Config.PprofEnabled`? Where are the endpoints served from in that case?

Reply:
Oh, looks like you're adding the pprof endpoints to `h.mux`, then using that later.

Reply:
Endpoints are served from [here](https://github.com/influxdata/influxdb/pull/15222/files#diff-719dcaf997cf7e555e32df7f85e7cd60R439) when any of the three properties are not true. This keeps existing behavior for all cases except when configured. I'll document that in the config description.

Reply:
Yup, endpoints are served from the existing handlers [here](https://github.com/influxdata/influxdb/pull/15222/files#diff-719dcaf997cf7e555e32df7f85e7cd60R439) when any of the three properties are not true. This keeps existing behavior for all cases except when configured. I'll add a better explanation in the example config.